src/components/Presenter.js

```diff
@@ -0,0 +1,358 @@
+
+const ItemStatus = {
+  INIT: 0,
+  LOADED: 1,
+  ENTERING: 2,
+  ENTERED: 3,
+  LEAVING: 4,
+  LEFT: 5,
+};
+
+export const PresenterDirection = {
+  RIGHT: 2,
+  LEFT: 3,
+};
+
+const TransitionHorizontalSlide = {
+  // STYLE TO DISPLAY THE SLIDE
+  normal: {
+    opacity: 1,
+    left: 0,
+    right: 0,
+  },
+
+  // STYLE TO START SLIDE
+  enter: {
+    opacity: 0,
+    left: '-100%',
+    right: '100%',
+  },
+
+  // STYLE TO FINISH SLIDE
+  left: {
+    opacity: 0,
+    left: '100%',
+    right: '-100%',
+  },
+};
+
+class PresenterItem extends React.Component {
+  render() {
+    const { item, transitionDuration, enterStyle, stateStyle } = this.props;
+    if (item) {
+      return (
+        <div style={stateStyle} className="item">
+          <img src={item.image} />
+          <div className="text">
+            <p>{item.text}</p>
+          </div>
+        </div>
+      );
+    } else return null;
+  }
+}
+
+class Presenter extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      index: -1,
+      status: PresenterStatus.INIT,
+      current: null,
+      direction: props.direction,
+    };
+    this.timerChange = null;
+
+    this.setupNode = this.setupNode.bind(this);
+
+    this.handleStart = this.handleStart.bind(this);
+    this.handleEnd = this.handleEnd.bind(this);
+    this.handlePausePlay = this.handlePausePlay.bind(this);
+    this.handleNodeAnimEnd = this.handleNodeAnimEnd.bind(this);
+
+    this.handleNext = this.handleNext.bind(this);
+    this.handlePrev = this.handlePrev.bind(this);
+  }
+
+  componentDidMount() {
+    const { startTimeout } = this.props;
+    this.setState({
+      status: PresenterStatus.LOADED,
+    });
+    this.timerStart = setTimeout(this.handleStart, startTimeout);
+  }
+
+  handleStart() {
+    const next = this.loadNextItem();
+    this.setState({
+      status: PresenterStatus.PLAYING,
+      index: next.index,
+      current: {
+        item: next.item,
+        status: ItemStatus.LOADED,
+      },
+    });
+  }
+
+  handleNext() {
+    const next = this.loadNextItem(PresenterDirection.RIGHT);
+    if (this.timerChange) {
+      clearTimeout(this.timerChange);
+      this.timerChange = null;
+    }
+
+    this.setState({
+      index: next.index,
+      direction: PresenterDirection.RIGHT,
+      current: {
+        item: next.item,
+        status: ItemStatus.LOADED,
+      },
+    });
+  }
+
+  handlePrev() {
+    const next = this.loadNextItem(PresenterDirection.LEFT);
+    if (this.timerChange) {
+      clearTimeout(this.timerChange);
+      this.timerChange = null;
+    }
+
+    this.setState({
+      index: next.index,
+      direction: PresenterDirection.LEFT,
+      current: {
+        item: next.item,
+        status: ItemStatus.LOADED,
+      },
+    });
+  }
+
+  loadNextItem(dir = null) {
+    const { current, index, direction } = this.state;
+    const { items } = this.props;
+
+    var nidx = index;
+
+    const d = dir ? dir : direction;
```

Comment:
you can use shortcircuting here
`const d = dir || direction;`
Please use meaningful/readable variable names. Avoid to use just `d`